controller_connections.go

```diff
@@ -40,6 +40,9 @@ func (c *controller) manageOnline() {
 				c.net.prom.Incoming.Set(float64(c.peers.Incoming()))
 				c.net.prom.Outgoing.Set(float64(c.peers.Outgoing()))
 			}
+		case <-c.net.stopper:
+			// ordered second so that manageOnline will clear out all peers from the peer store first
```

Comment:
Are you saying `c.peerStatus` will be drained before going to the stopper? Are you sure?

https://www.reddit.com/r/golang/comments/4ibs68/why_is_the_select_statement_nondeterministic/
https://stackoverflow.com/questions/11117382/priority-in-go-select-statement-workaround

Reply:
I was not aware of that, thank you!